### PR TITLE
Improvements to logging system

### PR DESCRIPTION
**Pull request by mkutlak**
I did this mainly because it was not possible to make any changes to format of logging messages without editing the code.
 
___
**Changes**
- Logging configuration from a file (`/etc/faf/faf-logging.conf`)
- Added DATETIME to logging messages:
```
[DATE TIME] LEVEL:NAME: MESSAGE
```
- Added THREADNAME for logging from threads:
```
[DATE TIME] LEVEL:NAME(THREADNAME): MESSAGE
```
- Lazy formatting of debugging logs
- Removed custom logger for Python 2.6
- Get indexes using enumerate in for loops
- Added some debugging messages 
- Compute len of arrays outside of for loops

The old format:
```
LEVEL:NAME: MESSAGE
```

Also see: https://docs.python.org/3/library/logging.html
Related to #806

### CODE REVIEW

**Comment by mkutlak** [comment]
Let me know if you have more suggestions what could be improved/changed for logging.

**Comment by ernestask** [comment]
Just nitpicks right now: using tabs instead of manually banging out four or however many spaces.

File: src/pyfaf/config.py [hunk 1]
```diff
@@ -87,6 +90,24 @@ def load_config():
...
+    """
+    Load main config file for logging.
+    """
+    main_log_config_file = None
+    fpath = os.path.join(MAIN_CONFIG_DIR, MAIN_LOG_CONFIG_FILE)
+
+    if CONFIG_LOG_FILE_ENV_VAR in os.environ:
+        fpath = os.environ[CONFIG_LOG_FILE_ENV_VAR]
+
+    if os.access(fpath, os.R_OK):
```
Comment (ernestask) [inline src/pyfaf/config.py]: One thing is that these checks are prone to TOCTOU bugs. Even https://docs.python.org/3.7/library/os.html#os.access provides a better alternative.

File: src/pyfaf/actions/create_problems.py [hunk 2]
```diff
@@ -299,8 +294,7 @@ def _create_problems(self, db, problemplugin, #pylint: disable=too-many-statemen
             i = 0
             for db_report in db_reports:
                 i += 1
```
Comment (ernestask) [inline src/pyfaf/actions/create_problems.py]: In most cases like this, we could just do `for i, db_report in enumerate(db_reports, start=1)`.

**Comment by ernestask** [comment]
Oh, and doing `len(whatever)` just for log messages in a loop can be slow, but doesn’t matter a whole lot from what I’ve observed.

**Comment by ernestask** [comment]
I’ll add some links to places where we could add messages (which I did in prod, but those don’t survive too long) that are rather helpful with timestamps now that we have them.

**Comment by ernestask** [comment]
For example:
https://github.com/abrt/faf/blob/ff904405c519b895e3ac58de7192cda5b8ec4d41/src/pyfaf/actions/create_problems.py#L265
https://github.com/abrt/faf/blob/ff904405c519b895e3ac58de7192cda5b8ec4d41/src/pyfaf/actions/create_problems.py#L266

https://github.com/abrt/faf/blob/ff904405c519b895e3ac58de7192cda5b8ec4d41/src/pyfaf/problemtypes/core.py#L226

I just added a bunch of “[problem plugin] Fooing bars” before any large-ish SQL query (or before handing something off to satyr). Printing the problem plugin is also not a bad idea, since we run a bunch of processes that write to the same file, so figuring out what’s what can be a bit of a game.

**Comment by ernestask** [comment]
Otherwise I don’t have any more suggestions, this is great work!

File: config/Makefile.am [hunk 3]
```diff
@@ -30,4 +31,5 @@ EXTRA_DIST = celery-beat-env.conf \
              celery-worker-env.conf \
              faf.conf.in \
              faf-web.conf.in \
+             faf-logging.conf \
```
Comment (mgrabovsky) [inline config/Makefile.am]: Just a teeny tiny suggestion: Move the added line one up so that the list is alphabetically sorted.

Comment (ernestask) [reply config/Makefile.am]: I approve this message.

**Comment by mkutlak** [comment]
> Just nitpicks right now: using tabs instead of manually banging out four or however many spaces.

Done.

> Oh, and doing `len(whatever)` just for log messages in a loop can be slow, but doesn’t matter a whole lot from what I’ve observed.

Done.

> I’ll add some links to places where we could add messages (which I did in prod, but those don’t survive too long) that are rather helpful with timestamps now that we have them.

Done.

File: src/pyfaf/config.py [hunk 4]
```diff
@@ -87,6 +89,29 @@ def load_config():
 
     return result
 
+def configure_logging():
+    """
+    Load and configure logging from a config file.
+    If the config file isn't available, fallback to basic logging.
```
Comment (ernestask) [inline src/pyfaf/config.py]: __***Extreme***__ nitpick: “fallback” is a noun, what you mean is “fall back”.

**Comment by ernestask** [comment]
I’ll merge this a bit later.

**Pull request merged**